src/backend/optimizer/path/allpaths.c

```diff
@@ -235,6 +247,23 @@ static void
 set_base_rel_sizes(PlannerInfo *root)
 {
 	Index		rti;
+	int			resultRelIndex = root->parse->resultRelation;
+	RelOptInfo *resultRel;
+
+	/*
+	 * If result relation is excluded by constraint exclusion we shouldn't
+	 * plan this sub-query
```

Comment:
Please, add the notice about SELECT stmts that in this case resultRelIndex is zero, resultRel is NULL and all successive branch is omitted.